OneCommunityGlobal/HighestGoodNetworkApp
# Vijay Anand - Unit test notificationReducer file

# Description
Added unit tests for the notificationReducer.

## Related PRS (if any):
No related PRs.

## Main changes explained:

- Added a new test file notificationReducer.test.js under the __tests__ directory.
- Test cases include:
- Returns the initial state when no state is provided.
  - Handles FETCH_USER_NOTIFICATIONS_REQUEST by setting loading to true and clearing error.
  - Handles FETCH_USER_NOTIFICATIONS_SUCCESS by populating the notifications array and setting loading to false.
  - Handles FETCH_USER_NOTIFICATIONS_FAILURE by updating the error field with the provided error message.
  - Handles FETCH_UNREAD_USER_NOTIFICATIONS_SUCCESS by updating the unreadNotifications array.
  - Handles MARK_NOTIFICATION_AS_READ_SUCCESS by removing the specified notification from the unreadNotifications array.
  - Handles RESET_ERROR by clearing the error field.

## How to test:

1. Check out the current branch.
2. Run npm install if necessary.
3. Execute npm test notificationReducer.test.js.
4. Verify that all test cases pass without any errors.

## Screenshots or videos of changes:
<img width="1014" alt="Screenshot 2024-11-26 at 11 08 44 PM" src="https://github.com/user-attachments/assets/de12f567-1046-47d0-89eb-ee67a8f292b1">

## Note:
Include the information the reviewers need to know.
